src/sorting/bogo_sort.rs

```diff
@@ -0,0 +1,73 @@
+use crate::math::PCG32;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+const DEFAULT: u64 = 4294967296;
+
+fn is_sorted<T: Ord>(arr: &[T], len: usize) -> bool {
+    for i in 0..len - 1 {
+        if arr[i] > arr[i + 1] {
+            return false;
+        }
+    }
+
+    true
+}
+
+#[cfg(target_pointer_width = "64")]
+fn generate_index(range: usize, generator: &mut PCG32) -> usize {
+    generator.get_u64() as usize % range
+}
+
+#[cfg(not(target_pointer_width = "64"))]
+fn generate_index(range: usize, generator: &mut PCG32) -> usize {
+    generator.get_u32() as usize % range
+}
+
+/**
+ * Fisher–Yates shuffle for generating random permutation.
+ */
+fn permute_randomly<T>(arr: &mut [T], len: usize, generator: &mut PCG32) {
+    for i in (1..len).rev() {
+        let j = generate_index(i + 1, generator);
+        arr.swap(i, j);
+    }
+}
+
+pub fn bogo_sort<T: Ord>(arr: &mut [T]) {
```

Comment:
Have you considered allowing for a 'seeded' bogosort, simply to ensure consistency in testing?

Reply:
What do you mean by consistency? The execution time? It's not something that is tested and the end result is the same regardless of the way it's obtained